File: Python/programmers/etc/level1/체육복.py
def solution(n, lost, reserve):
    sr = set(reserve)-set(lost)
    sl = set(lost)-set(reserve)
    for i in sr:
        if i-1 in sl:
            sl.remove(i-1)
        elif i+1 in sl:
            sl.remove(i+1)
    return n-len(sl)


n = 5
lost, reserve = [2, 4], [1, 3, 5]
print(solution(n, lost, reserve)) # 5

n = 5
lost, reserve = [2, 4], [3]
print(solution(n, lost, reserve)) # 4

n = 3
lost, reserve = [3], [1]
print(solution(n, lost, reserve)) # 2

File: Java/programmers/etc/level1/체육복.java
package CodeTest.Java.programmers.etc.level1;

public class 체육복 {
    public int solution(int n, int[] lost, int[] reserve) {
        int answer = n;
        int[] people = new int[n];
        
        for (int l : lost) people[l-1]--;
        for (int r : reserve) people[r-1]++;
        
        for (int i=0; i<n; i++) {
            if (people[i] == -1) {
                if (i-1 >= 0 && people[i-1] == 1) {
                    people[i]++; people[i-1]--;
                } else if (i+1 < n && people[i+1] == 1) {
                    people[i]++; people[i+1]--;
                } else {
                    answer--;
                }
            }
        }
        
        return answer;
    }
}
